randovania/generator/dock_weakness_distributor.py

```diff
@@ -0,0 +1,193 @@
+
+        if patches.configuration.dock_rando.mode == DockRandoMode.ONE_WAY:
+            player_docks.extend(((player, identifier) for identifier in patches.dock_weakness.keys()))
+        
+        if patches.configuration.dock_rando.mode == DockRandoMode.TWO_WAY:
+            game = descriptions[player]
+            ctx = NodeContext(
+                patches,
+                patches.starting_items,
+                game.resource_database,
+                game.world_list
+            )
+
+            for identifier in patches.dock_weakness:
+                dock: DockNode = game.world_list.node_by_identifier(identifier)
+                if (player, dock.get_target_identifier(ctx)) not in player_docks:
+                    player_docks.append((player, identifier))
+        
+        if TO_SHUFFLE_PROPORTION < 1.0:
+            rng.shuffle(player_docks)
+            limit = int(len(player_docks) * TO_SHUFFLE_PROPORTION)
+            player_docks = player_docks[:limit]
+        
+        unassigned_docks.extend(player_docks)
+    
+    rng.shuffle(unassigned_docks)
+
+    docks_placed = 0
+    docks_to_place = len(unassigned_docks)
+
+    start_time = time.perf_counter()
+
+    while unassigned_docks:
+        await asyncio.sleep(0)
+        status_update(f"{docks_placed}/{docks_to_place} docks placed")
+
+        player, identifier = unassigned_docks.pop()
+
+        game = descriptions[player]
+        patches = all_patches[player]
+
+        dock: DockNode = game.world_list.node_by_identifier(identifier)
+        target: DockNode = game.world_list.node_by_identifier(dock.get_target_identifier(NodeContext(
+            patches,
+            patches.starting_items,
+            game.resource_database,
+            game.world_list,
+        )))
+
+        dock_type_params = game.dock_weakness_database.dock_rando_params[dock.dock_type]
+        dock_type_state = patches.configuration.dock_rando.types_state[dock.dock_type]
+
+        # Calculate the reach assuming this dock is completely locked
+        locks = {identifier: dock_type_params.locked}
+        if patches.configuration.dock_rando.mode == DockRandoMode.TWO_WAY:
+            locks[target.identifier] = dock_type_params.locked
+        locked_patches = patches.assign_dock_weakness_assignment(locks)
+
+        state, logic = resolver.setup_resolver(patches.configuration, locked_patches)
+        logic = DockRandoLogic.from_logic(logic, dock)
+        debug.log_resolve_start()
+        debug.debug_print(f"{identifier}")
+        
+        level = debug.debug_level()
+        debug.set_level(0)
+
+        try:
+            new_state = await resolver.advance_depth(state, logic, lambda s: None, max_attempts=RESOLVER_ATTEMPTS)
+        except resolver.ResolverTimeout:
+            new_state = None
+            result = f"Timeout ({debug.get_attempts()} attempts)"
+        else:
+            result = f"Finished resolver ({debug.get_attempts()} attempts)"
```

Comment:
Could this `result = "Finished ..."` be just after the `await resolver.advance_depth` call inside the `try` block?

Reply:
it could; I thought I was supposed to only have the part that expects the exception in the try block but I can change that no problem

Reply:
Oh is that the expected use of `else` in a try/except? hm

Reply:
that's what I've read yeah

Reply:
🤷 I'm fine either way